lib/checks/report/report.rb

```diff
@@ -0,0 +1,28 @@
+module Checks
+  module Report
+    class Report
```

Comment:
Maybe. I hope there is a better alternative.
I don't really want anything other than checker.rb and check_runner.rb in lib (the progress_reporter thing is its own concern).
The Report class is a common feature of checks and belongs somewhere under the checks dir and in the Checks namespace.
I moved it from being a direct entry in the check dir as this makes loading all checks by default easier.